webapi/src/test/scala/dsp/valueobjects/KnoraProjectSpec.scala

```diff
@@ -0,0 +1,182 @@
+  def spec = suite("ProjectSpec")(
+    shortcodeTest,
+    shortnameTest,
+    nameTest,
+    descriptionTest,
+    keywordsTest,
+    logoTest,
+    projectStatusTest,
+    projectSelfJoinTest
+  )
+
+  private val shortcodeTest = suite("Shortcode")(
+    test("pass an empty value and return an error") {
+      assertTrue(
+        Shortcode.from("") == Validation.fail(ValidationException("Shortcode cannot be empty."))
+      )
+    },
+    test("pass an invalid value and return an error") {
+      val invalidShortcodes = Gen.fromIterable(Seq("123", "000G", "12345"))
+      check(invalidShortcodes) { shortcode =>
+        assertTrue(
+          Shortcode.from(shortcode) == Validation.fail(ValidationException(s"Shortcode is invalid: $shortcode"))
+        )
+      }
+    },
+    test("pass a valid value and successfully create value object") {
+      assertTrue(Shortcode.from("FFFF").map(_.value).contains("FFFF"))
+    }
+  )
+
+  private val shortnameTest = suite("Shortname")(
+    test("pass an empty value and return validation error") {
+      assertTrue(
+        Shortname.from("") == Validation.fail(ValidationException("Shortname cannot be empty."))
+      )
+    },
+    test("pass invalid values and return validation error") {
+      val gen = Gen.fromIterable(
+        Seq(
+          "invalid:shortname",
+          "-invalidshortname",
+          "_invalidshortname",
+          ".invalidshortname",
+          "invalid/shortname",
+          "invalid@shortname",
+          "1invalidshortname",
+          " invalidshortname",
+          "invalid shortname",
+          "a",
+          "ab",
+          "just-21char-shortname"
+        )
+      )
+      check(gen) { param =>
+        assertTrue(
+          Shortname.from(param) == Validation.fail(ValidationException(s"Shortname is invalid: $param"))
+        )
+      }
+    },
+    test("pass valid values and successfully create value objects") {
+      val gen = Gen.fromIterable(
+        Seq(
+          "DefaultSharedOntologiesProject",
+          "valid-shortname",
+          "valid_shortname",
+          "validshortname-",
+          "validshortname_",
+          "abc",
+          "a20charLongShortname"
+        )
+      )
+      check(gen)(param => assertTrue(Shortname.from(param).map(_.value) == Validation.succeed(param)))
+    }
+  )
+
+  private val nameTest = suite("Name")(
+    test("pass invalid Name and expect an error to be returned") {
+      val invalidNames =
+        Gen.stringBounded(0, 2)(Gen.printableChar) ++ Gen.stringBounded(257, 300)(Gen.printableChar)
+      check(invalidNames) { name =>
+        assertTrue(
+          Longname.from(name) == Validation.fail(ValidationException("Longname must be 3 to 256 characters long."))
+        )
+      }
+    },
+    test("pass a valid value and successfully create value object") {
+      val validNames = Gen.stringBounded(3, 256)(Gen.printableChar)
+      check(validNames) { name =>
```

Comment:
```suggestion
  private val nameTest = suite("Longname")(
    test("pass invalid Longname and expect an error to be returned") {
      val invalidLongnames =
        Gen.stringBounded(0, 2)(Gen.printableChar) ++ Gen.stringBounded(257, 300)(Gen.printableChar)
      check(invalidLongnames) { name =>
        assertTrue(
          Longname.from(name) == Validation.fail(ValidationException("Longname must be 3 to 256 characters long."))
        )
      }
    },
    test("pass a valid value and successfully create value object") {
      val validLongnames = Gen.stringBounded(3, 256)(Gen.printableChar)
      check(validLongnames) { name =>
```

Reply:
I have added d60a54f7aadc9b475bf4f0b9100799436f1cf785 to the other PR, which should clarify the naming a bit.